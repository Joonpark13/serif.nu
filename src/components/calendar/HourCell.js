import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import {
  sectionsSelector,
  sectionsForHourSelector,
  associatedClassesForHourSelector,
  sectionPreviewSelector,
  associatedClassPreviewSelector,
  allSectionPreviewsSelector,
} from 'selectors';
import useSelector from 'util/use-selector';
import { columnBorderStyle, cellMinHeight } from './calendar-constants';
import CalendarSection from './CalendarSection';
import AssociatedClass from './AssociatedClass';

export const styles = {
  calendarCell: {
    flexGrow: 1,
    borderBottom: columnBorderStyle,
    minHeight: cellMinHeight,
    position: 'relative',
  },
};

function HourCell({ hour, dow, classes }) {
  const sections = useSelector(state => sectionsForHourSelector(state, hour, dow));
  const associatedClasses = useSelector(
    state => associatedClassesForHourSelector(state, hour, dow),
  );
  const sectionPreview = useSelector(state => sectionPreviewSelector(state, hour, dow));
  const associatedClassPreview = useSelector(
    state => associatedClassPreviewSelector(state, hour, dow),
  );
  const allSections = useSelector(sectionsSelector);
  const allSectionPreviews = useSelector(allSectionPreviewsSelector);

  return (
    <div className={classes.calendarCell}>
      {sections.map(section => (
        <CalendarSection key={section.id} section={section} />
      ))}
      {associatedClasses.map(associatedClass => (
        <AssociatedClass
          key={JSON.stringify(associatedClass.event)}
          associatedClass={associatedClass}
          section={allSections.find(section => section.id === associatedClass.sectionId)}
        />
      ))}
      {sectionPreview && <CalendarSection section={sectionPreview} isPreview />}
      {associatedClassPreview && (
        <AssociatedClass
          isPreview
          associatedClass={associatedClassPreview}
          section={allSectionPreviews[0]}
        />
      )}
    </div>
  );
}

HourCell.propTypes = {
  hour: PropTypes.number.isRequired,
  dow: PropTypes.string.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export { HourCell as UnstyledHourCell };

export default withStyles(styles)(HourCell);
