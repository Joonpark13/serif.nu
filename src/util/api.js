/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
