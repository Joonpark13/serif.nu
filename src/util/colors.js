export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
