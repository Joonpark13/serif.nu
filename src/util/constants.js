/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
