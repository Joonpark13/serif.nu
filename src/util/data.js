import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
