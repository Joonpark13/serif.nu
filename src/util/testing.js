/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
