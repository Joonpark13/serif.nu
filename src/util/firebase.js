import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
