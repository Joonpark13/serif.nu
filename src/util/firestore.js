import firebase from 'firebase/app';
import 'firebase/firestore';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export default firebase.firestore();
