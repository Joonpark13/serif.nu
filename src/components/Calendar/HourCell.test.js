import React from 'react';
import { shallow } from 'enzyme';
import { mockStyles } from 'util/testing';
import { UnstyledHourCell, styles } from './HourCell';

describe('HourCell', () => {
  it('renders correctly', () => {
    const classes = mockStyles(styles);
    const testSections = [{ id: 12345 }];
    const wrapper = shallow(
      <UnstyledHourCell hour={10} dow="Mo" sections={testSections} classes={classes} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
