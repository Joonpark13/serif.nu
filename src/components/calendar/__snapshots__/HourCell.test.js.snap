// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourCell renders correctly 1`] = `
<div
  className="makeStyles-calendarCell-1"
>
  <CalendarSection
    isPreview={false}
    section={
      Object {
        "id": "12345",
      }
    }
  />
  <AssociatedClass
    associatedClass={
      Object {
        "event": Object {},
        "sectionId": "12345",
      }
    }
    isPreview={false}
    section={
      Object {
        "id": "12345",
      }
    }
  />
</div>
`;
